### PR TITLE
workaround for nextTick issue in Edge

### PR DESCRIPTION
**Pull request by brettsam**
Addresses #223 and #241.

### CODE REVIEW

File: src/WebJobs.Script/Description/Node/Script/globalInitialization.js [hunk 1]
```diff
@@ -8,6 +8,9 @@ return function (context, callback) {{
         context.handleUncaughtException(err.stack);
     }});
 
+    // workaround for https://github.com/tjanczuk/edge/issues/325
+    process.nextTick = global.setImmediate;
```
Comment (mathewc) [inline src/WebJobs.Script/Description/Node/Script/globalInitialization.js]: Please put a TEMP HACK comment on this so we remove it. Also recommend logging a new issue in our repo to clean this up.

File: test/WebJobs.Script.Tests/TestScripts/Node/NextTick/function.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿{
+    "bindings": [
```
Comment (mathewc) [inline test/WebJobs.Script.Tests/TestScripts/Node/NextTick/function.json]: Can you just add this as a new branch in the Scenarios function? See the pattern I've been employing. You can then add a test that does a manual invoke to make sure it doesn't hang.

Comment (brettsam) [inline test/WebJobs.Script.Tests/TestScripts/Node/NextTick/function.json]: Done.

File: src/WebJobs.Script/Description/Node/Script/globalInitialization.js (hunk 1 above)
Comment (brettsam) [inline src/WebJobs.Script/Description/Node/Script/globalInitialization.js]: Done.

File: src/WebJobs.Script/Description/Node/Script/globalInitialization.js [hunk 3]
```diff
@@ -8,6 +8,9 @@ return function (context, callback) {{
         context.handleUncaughtException(err.stack);
     }});
 
+    // TEMP HACK: workaround for https://github.com/tjanczuk/edge/issues/325
```
Comment (mathewc) [inline src/WebJobs.Script/Description/Node/Script/globalInitialization.js]: Please also add a new issue in this repo to remove this hack, referencing the Edge bug

**Comment by mathewc** [comment]
:ship

File: src/WebJobs.Script/Description/Node/Script/globalInitialization.js (hunk 3 above)
Comment (brettsam) [inline src/WebJobs.Script/Description/Node/Script/globalInitialization.js]: Done.

---
In reply to: [70371964](https://github.com/Azure/azure-webjobs-sdk-script/pull/486#discussion_r70371964) [](ancestors = 70371964)

Comment (danderson00) [inline src/WebJobs.Script/Description/Node/Script/globalInitialization.js]: This is really bad. There are differences between process.nextTick and global.setImmediate, you could be breaking third party packages in really, nasty, subtle ways.

Comment (mathewc) [inline src/WebJobs.Script/Description/Node/Script/globalInitialization.js]: the current state of affairs is that ANY calls to process.nextTick currently do not succeed. So with this change we're no worse off. any packages using nexttick currently do not work.

**Pull request merged**